### DISCUSSION TITLE
How to control speed of animation for animated edges

### DISCUSSION THREAD

**Discussion by ghost** [xyflow/xyflow#2994] [Q&A] [answered]
I tried animated edge, and I want to speed up or down the animation. How to do it

**Comment by bcakmakoglu** [comment, marked as answer]
You could just use css to create different animation speeds or use `style` on certain edges, for example

```js
    style: {
      // use 0.2s to speed up the animation
      animation: "dashdraw 0.2s linear infinite",
      strokeDasharray: 5
    }
```

[Here's a sandbox.](https://codesandbox.io/s/elated-davinci-onbriy?file=/initial-elements.js:1675-1768)

**Reply by ghost** [reply]
how to control this animation for all the edges from a single input ?

Solution I have is that, I'll listen to a variable change, and on change, Ill update the animation speed of all the edges through the style property.

But, if I update the state, it will reflect only when I change the all edges and pass it to react-flow ?

**Reply by bcakmakoglu** [reply]
Really depends, with a custom edge you could setup something differently instead of updating the whole state but using the default edges you'll probably have to do something like this

```ts
setEdges((edges) => edges.map(edge => ({ ...edge, style: animationStyle }))
```

**Comment by parasLR** [comment]
Hi, Guys. I need to change the animated dash to the animated arrow. 

Can I do that? Please help me.
